homeassistant/components/aprilaire/select.py

```diff
@@ -0,0 +1,157 @@
+                translation_key="air_cleaning_event",
+                options_map=AIR_CLEANING_EVENT_MAP,
+                event_value_key=Attribute.AIR_CLEANING_EVENT,
+                mode_value_key=Attribute.AIR_CLEANING_MODE,
+                is_event=True,
+                select_option_fn=coordinator.client.set_air_cleaning,
+            )
+        )
+
+        descriptions.append(
+            AprilaireSelectDescription(
+                key="air_cleaning_mode",
+                translation_key="air_cleaning_mode",
+                options_map=AIR_CLEANING_MODE_MAP,
+                event_value_key=Attribute.AIR_CLEANING_EVENT,
+                mode_value_key=Attribute.AIR_CLEANING_MODE,
+                is_event=False,
+                select_option_fn=coordinator.client.set_air_cleaning,
+            )
+        )
+
+    if coordinator.data.get(Attribute.VENTILATION_AVAILABLE) == 1:
+        descriptions.append(
+            AprilaireSelectDescription(
+                key="fresh_air_event",
+                translation_key="fresh_air_event",
+                options_map=FRESH_AIR_EVENT_MAP,
+                event_value_key=Attribute.FRESH_AIR_EVENT,
+                mode_value_key=Attribute.FRESH_AIR_MODE,
+                is_event=True,
+                select_option_fn=coordinator.client.set_fresh_air,
+            )
+        )
+
+        descriptions.append(
+            AprilaireSelectDescription(
+                key="fresh_air_mode",
+                translation_key="fresh_air_mode",
+                options_map=FRESH_AIR_MODE_MAP,
+                event_value_key=Attribute.FRESH_AIR_EVENT,
+                mode_value_key=Attribute.FRESH_AIR_MODE,
+                is_event=False,
+                select_option_fn=coordinator.client.set_fresh_air,
+            )
+        )
+
+    entities = (
+        AprilaireSelectEntity(coordinator, description, config_entry.unique_id)
+        for description in descriptions
+    )
+
+    async_add_entities(entities)
+
+
+@dataclass(frozen=True, kw_only=True)
+class AprilaireSelectDescription(SelectEntityDescription):
+    """Class describing Aprilaire select entities."""
+
+    options_map: dict[int, str]
+    event_value_key: str
+    mode_value_key: str
+    is_event: bool
+    select_option_fn: Callable[[int, int], Awaitable]
+
+
+class AprilaireSelectEntity(BaseAprilaireEntity, SelectEntity):
+    """Base sensor entity for Aprilaire."""
+
+    entity_description: AprilaireSelectDescription
+
+    def __init__(
+        self,
+        coordinator: AprilaireCoordinator,
+        description: AprilaireSelectDescription,
+        unique_id: str,
+    ) -> None:
+        """Initialize a sensor for an Aprilaire device."""
+
+        self.entity_description = description
+        self.values_map = {v: k for k, v in description.options_map.items()}
+
+        super().__init__(coordinator, unique_id)
+
+        self._attr_options = list(set(description.options_map.values()))
```

Comment:
The `list(set(..))` was because in the sensor there were a lot of duplicate options, but in this case everythin is unique, so no need to do this now

Reply:
Updated